File: python ver/camera.py
import cv2

class Camera:
  cam = 0
  frame = 0
  def __init__(self):
    self.cam = cv2.VideoCapture(0)
    self.cam.set(cv2.CAP_PROP_FPS, 24)
    self.cam.set(cv2.CAP_PROP_BUFFERSIZE, 1)
  
  def __init__(self, video_num):
    self.cam = cv2.VideoCapture(int(video_num))
    self.cam.set(cv2.CAP_PROP_FPS, 24)
    self.cam.set(cv2.CAP_PROP_BUFFERSIZE, 1)
  
  def resize_frame(self, frame_width, frame_height):
    self.cam.set(cv2.CAP_PROP_FRAME_WIDTH, int(frame_width))
    self.cam.set(cv2.CAP_PROP_FRAME_HEIGHT, int(frame_height))
  
  def get_frame(self):
    ret, self.frame = self.cam.read()
    return self.frame
  
  def show_frame(self):
    cv2.imshow("frame", self.frame)
    cv2.waitKey(1)

  def shutdown(self):
    cam.cv2.release()

File: cpp ver/camera.cpp
#include "camera.h"

Camera::Camera() {
  cam = cv::VideoCapture(0);
  cam.set(cv::CAP_PROP_FPS, 24);
  cam.set(cv::CAP_PROP_BUFFERSIZE, 1);
}

Camera::Camera(int video_num) {
  cam = cv::VideoCapture(video_num);
  cam.set(cv::CAP_PROP_FPS, 24);
  cam.set(cv::CAP_PROP_BUFFERSIZE, 1);
}

void Camera::resize_frame(int frame_width, int frame_height) {
  cam.set(cv::CAP_PROP_FRAME_WIDTH, frame_width);
  cam.set(cv::CAP_PROP_FRAME_HEIGHT, frame_height);
}

cv::Mat Camera::get_frame(void) {
  cam.read(frame);
  return frame;
}

void Camera::shundown(void) {
  cam.cv::release();
}

void Camera::show_frame(void) {
  cv::imshow("frame", frame);
  cv::waitKey(1);
}

